src/main/java/org/threeten/extra/chrono/InternationalFixedChronology.java

```diff
@@ -0,0 +1,484 @@
+     * @return the calendar system era, not null
+     * @throws DateTimeException if unable to create the era
+     */
+    @Override
+    public InternationalFixedEra eraOf(final int eraValue) {
+        return InternationalFixedEra.of(eraValue);
+    }
+
+    /**
+     * Gets the list of eras for the chronology.
+     * <p>
+     * Most calendar systems have an era, within which the year has meaning.
+     * If the calendar system does not support the concept of eras, an empty
+     * list must be returned.
+     *
+     * @return the list of eras for the chronology, may be immutable, not null
+     */
+    @Override
+    public List<Era> eras() {
+        return Arrays.<Era>asList(InternationalFixedEra.values());
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ValueRange range(final ChronoField field) {
+        switch (field) {
+            case ALIGNED_DAY_OF_WEEK_IN_YEAR:
+            case ALIGNED_DAY_OF_WEEK_IN_MONTH:
+            case DAY_OF_WEEK:
+                return DAY_OF_WEEK_RANGE;
+            case ALIGNED_WEEK_OF_MONTH:
+                return WEEK_OF_MONTH_RANGE;
+            case ALIGNED_WEEK_OF_YEAR:
+                return WEEK_OF_YEAR_RANGE;
+            case DAY_OF_MONTH:
+                return DAY_OF_MONTH_RANGE;
+            case DAY_OF_YEAR:
+                return DAY_OF_YEAR_RANGE;
+            case EPOCH_DAY:
+                return EPOCH_DAY_RANGE;
+            case ERA:
+                return ERA_RANGE;
+            case MONTH_OF_YEAR:
+                return MONTH_OF_YEAR_RANGE;
+            case PROLEPTIC_MONTH:
+                return PROLEPTIC_MONTH_RANGE;
+            case YEAR_OF_ERA:
+            case YEAR:
+                return YEAR_RANGE;
+            default:
+                return field.range();
+        }
+    }
+
+    /**
+     * Calculates the proleptic-year given the era and year-of-era.
+     * <p>
+     * The International Fixed calendar only knows common era years, thus negative years are invalid.
+     * <p>
+     * If the chronology makes active use of eras, such as {@code JapaneseChronology}
+     * then the year-of-era will be validated against the era.
+     * For other chronologies, validation is optional.
+     *
+     * @param era       the era of the correct type for the chronology, not null
+     * @param yearOfEra the chronology year-of-era
+     * @return the proleptic-year
+     * @throws DateTimeException  if unable to convert to a proleptic-year, such as if the year is invalid for the era
+     * @throws ClassCastException if the {@code era} is not of the correct type for the chronology
+     */
+    @Override
+    public int prolepticYear(final Era era, final int yearOfEra) {
+        if (!(era instanceof InternationalFixedEra)) {
+            throw new ClassCastException("Invalid era: " + era);
+        }
+
+        if (0 > yearOfEra) {
```

Comment:
Defensive programming.
In comparisons, the constant goes first and the variable follows.
Avoid assignment (= vs ==), because you cannot assign a value to a constant.
Though not pretty, almost the same byte-code is produced from it.
However, I see your point, it changing will I.